e2e/cli/e2e.bats

```diff
@@ -0,0 +1,101 @@
+bats_load_library "bats-assert"
+bats_load_library "bats-file"
+bats_load_library "bats-support"
+
+setup_file() {
+    export CLI_BIN="$(dirname "${CLI_BIN}")/../main.js"
```

Comment:
This feels janky. My mental model was that I can just depend on the `js_binary` and run it within bats, but that didn't work and I needed to add a dependency on node and then run the entrypoint file directly. But it works :man_shrugging:

Reply:
In some cases rules_js binaries need `BAZEL_BINDIR` to be set, not sure if that applies here or not.

Reply:
Playing around but not making much progress. I'll accept that this works for now and I'll come back to this later.